src/AutoRest.CSharp/LowLevel/Output/LowLevelRestClient.cs

```diff
@@ -39,12 +40,41 @@ public LowLevelRestClient(OperationGroup operationGroup, BuildContext<LowLevelOu
             ClientOptions = new ClientOptionsTypeProvider(_context);
         }
 
+        protected RequestConditionCollapseType GetRequestConditionCollapseType(IEnumerable<RequestParameter> requestParameters)
+        {
+            RequestConditionCollapseType ret = RequestConditionCollapseType.None;
+            int matchCondCount = 0;
+            int requestCondCount = 0;
+            foreach (RequestParameter requestParameter in requestParameters)
+            {
+                if (requestParameter.IsMatchConditionHeader())
+                {
+                    matchCondCount++;
+                    requestCondCount++;
+                } else if (requestParameter.IsRequestConditionHeader())
+                {
+                    requestCondCount++;
+                }
+            }
+
+            if (requestCondCount == 4)
+            {
+                ret = RequestConditionCollapseType.RequestConditionsCollapse;
+            } else if (matchCondCount == 2)
+            {
+                ret = RequestConditionCollapseType.MatchConditionsCollapse;
+            }
+            return ret;
```

Comment:
If these branches are not coupled with each other, it will be more clear if we just short cut these branches, like
```suggestion
            if (requestCondCount == 4)
            {
                return RequestConditionCollapseType.RequestConditionsCollapse;
            }
            if (matchCondCount == 2)
            {
                return RequestConditionCollapseType.MatchConditionsCollapse;
            }
            return RequestConditionCollapseType.None;
```